### ISSUE TITLE
xhr/open-during-abort-processing.htm test does not seems to match the XHR spec

### ISSUE THREAD

**Issue by cdumez** [web-platform-tests/wpt#10217]
xhr/open-during-abort-processing.htm test does not seems to match the XHR spec. It expects the following upload events to get fired:
    "upload.onabort 1",
    "upload.onloadend 1",

The first request looks like this:
```
client.open("POST", "resources/content.py")
client.send('abcd')
```

Thus, yes, the request initially has a body so the following set in send():

> If req’s body is null, set the upload complete flag.

Will not set the upload complete flag.

But then the test calls abort() in the loadstart event listener. After that the readystatechange event listener gets called and the test does:
```
log.push('onreadystatechange readyState before open() ' + client.readyState)
client.open("GET", "resources/content.py")
log.push('onreadystatechange readyState after open() ' + client.readyState)
client.send(null)
```

Because this is a 'GET' request, send() will set the request body to null and the upload complete flag WILL be set. The abort() steps will then keep going and it will fire the abort event via theses steps:
https://xhr.spec.whatwg.org/#request-error-steps

Because the upload complete flag is NOW set, I do not expect step 6 to fire any upload events. Yet, the test seems to expect those upload events.

My understanding is that this is the reason why this test is failing in Firefox. The test is passing in Chrome. I hit the issue when trying to align WebKit with the specification because upload events were no longer fired after my change.

**Comment by cdumez** [comment]
cc @annevk

**Comment by cdumez** [comment]
To summarize, if you look at those steps:
- https://xhr.spec.whatwg.org/#request-error-steps

Step 5 (Fire an event named readystatechange) causes JS to be run, which can call open() / send() on the XHR again. If this happens, it will potentially change the value of the "upload complete flag" / "upload listener flag" before we reach Step 6.

This is an issue because step 6 relies on the "upload complete flag" / "upload listener flag" to decide wether on not to fire the events.

**Comment by annevk** [comment]
I think you're right that we should change this test. @wisniewskit thoughts?

**Comment by wisniewskit** [comment]
Interestingly, I have a new re-implementation of the XHR spec for Firefox in progress, and it's not failing this test. I'll have to investigate this a bit more closely to see why it's passing.

**Comment by wisniewskit** [comment]
Funny enough, I had forgotten at the time I first read this issue that I had filed this bug already as #7933, and worked around it in my new XHR implementation. So yes, I agree with @cdumez, and the fix is already being considered in #7933. I'll close this as a duplicate. Sorry for not noticing sooner!
